cpp/benchmarks/string/experimental/stringview_compare.cu

```diff
@@ -288,6 +288,68 @@ std::pair<rmm::device_uvector<ArrowBinaryView>, rmm::device_buffer> create_sv_ar
 
   return std::pair{std::move(d_items), std::move(data_buffer)};
 }
+
+template <typename MapIterator>
+std::pair<rmm::device_uvector<ArrowBinaryView>, rmm::device_buffer> gather_sv_array(
+  rmm::device_uvector<ArrowBinaryView> const& d_items,
+  rmm::device_buffer const& data,
+  MapIterator begin,
+  cudf::size_type map_size,
+  rmm::cuda_stream_view stream)
+{
+  auto output   = rmm::device_uvector<ArrowBinaryView>(map_size, stream);
+  auto d_output = output.data();
+  thrust::gather(rmm::exec_policy(stream), begin, begin + map_size, d_items.data(), d_output);
+  // Although the above should be enough, in reality it is impractical to share a data buffer
+  // between two columns in libcudf. Sharing data in column_views is expected but a libcudf
+  // gather (all other libcudf APIs) would return a new column with newly owned (non-shared)
+  // data. At best, the data-buffer could be simply copied but it seems impractical in general
+  // to not compact the buffer and remove any unused data characters from it.
+  // Also, the compaction could help coalesce accessing adjacent row data in down stream calls.
+  //
+  // The rest of the code compacts the data buffer appropriately.
+
+  // record sizes of the long buffers (only single data buffer is supported in this benchmark)
+  auto offsets   = rmm::device_uvector<int32_t>(map_size + 1, stream);
```

Comment:
Should `int32_t` be `size_type` here, as we normally do for offsets? If this is matching to an Arrow spec, I might be wrong.
```suggestion
  auto offsets   = rmm::device_uvector<cudf::size_type>(map_size + 1, stream);
```